datareduction/datareducer.cpp

```diff
@@ -36,18 +36,145 @@ void initializeDataReducers(DataReducer * outputReducer, DataReducer * diagnosti
+                                                }
+                                                ));
+         continue;
+      }
       if(*it == "B") { // Bulk magnetic field at Yee-Lattice locations
          outputReducer->addOperator(new DRO::VariableB);
          continue;
       }
+      if(*it == "fs_BackgroundB") { // Static (typically dipole) magnetic field part
+         outputReducer->addOperator(new DRO::DataReductionOperatorFsGrid("fs_background_B",[](
+                      FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, 2>& perBGrid,
+                      FsGrid< std::array<Real, fsgrids::efield::N_EFIELD>, 2>& EGrid,
+                      FsGrid< std::array<Real, fsgrids::ehall::N_EHALL>, 2>& EHallGrid,
+                      FsGrid< std::array<Real, fsgrids::egradpe::N_EGRADPE>, 2>& EGradPeGrid,
+                      FsGrid< std::array<Real, fsgrids::moments::N_MOMENTS>, 2>& momentsGrid,
+                      FsGrid< std::array<Real, fsgrids::dperb::N_DPERB>, 2>& dPerBGrid,
+                      FsGrid< std::array<Real, fsgrids::dmoments::N_DMOMENTS>, 2>& dMomentsGrid,
+                      FsGrid< std::array<Real, fsgrids::bgbfield::N_BGB>, 2>& BgBGrid,
+                      FsGrid< std::array<Real, fsgrids::volfields::N_VOL>, 2>& volGrid,
+                                                         FsGrid< fsgrids::technical, 2>& technicalGrid)->std::vector<double> {
+                                                        
+                                                std::array<int32_t,3>& gridSize = technicalGrid.getLocalSize();
+                                                std::vector<double> retval(gridSize[0]*gridSize[1]*gridSize[2]*3);
+
+                                                // Iterate through fsgrid cells and extract boundary flag
+                                                for(int z=0; z<gridSize[2]; z++) {
+                                                  for(int y=0; y<gridSize[1]; y++) {
+                                                    for(int x=0; x<gridSize[0]; x++) {
+                                                        retval[3*(gridSize[1]*gridSize[0]*z + gridSize[0]*y + x)] =     (*BgBGrid.get(x,y,z))[fsgrids::BGBX];
+                                                        retval[3*(gridSize[1]*gridSize[0]*z + gridSize[0]*y + x) + 1] = (*BgBGrid.get(x,y,z))[fsgrids::BGBY];
+                                                        retval[3*(gridSize[1]*gridSize[0]*z + gridSize[0]*y + x) + 2] = (*BgBGrid.get(x,y,z))[fsgrids::BGBZ];
+                                                    }
+                                                  }
+                                                }
+                                                return retval;
+                                                }
+                                                ));
+         continue;
+      }
       if(*it == "BackgroundB") { // Static (typically dipole) magnetic field part
```

Comment:
We will soon remove a lot of field information from DCCRG. Would it make more sense to re-use the old names for fsgrid writing, since for example BGBX will be cleaned away soon.  

In general, should one always say the grid (fsg vlg), or should we only add vlg_xxx or fsg_xxxx when the variable is not on the native grid? 
e.g. 
fsg_rho since rho is not native on fsg,  BackgroundB from fsgrid since is it is native fsgrid, and vlg_Bvol since it is not a native vlasov grid parameter.